dts/arm/nxp/nxp_mcxw71.dtsi

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <arm/armv8-m.dtsi>
+#include <zephyr/dt-bindings/clock/scg_k4.h>
+#include <zephyr/dt-bindings/gpio/gpio.h>
+#include <zephyr/dt-bindings/pwm/pwm.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			compatible = "arm,cortex-m33f";
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			mpu: mpu@e000ed90 {
+				compatible = "arm,armv8m-mpu";
+				reg = <0xe000ed90 0x40>;
+			};
+		};
+	};
+
+	soc {
+		ctcm: sram@14000000 {
+			ranges = <0x0 0x14000000 DT_SIZE_K(16)>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			ctcm0: code_memory@0 {
+				compatible = "mmio-sram";
+				reg = <0x0 DT_SIZE_K(16)>;
+			};
+		};
+
+		stcm: sram@30000000 {
+			ranges = <0x0 0x30000000 DT_SIZE_K(112)>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			stcm0: system_memory@0 {
+				compatible = "mmio-sram";
+				reg = <0x0 DT_SIZE_K(64)>;
+			};
+
+			stcm1: system_memory@1a000 {
+				compatible = "zephyr,memory-region","mmio-sram";
+				reg = <0x1a000 DT_SIZE_K(8)>;
+				zephyr,memory-region = "RetainedMem";
+			};
+		};
+
+		smu2: sram@489C0000 {
+			ranges = <0x0 0x489C0000 DT_SIZE_K(40)>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+		};
+
+		peripheral: peripheral@50000000 {
+			ranges = <0x0 0x50000000 DT_SIZE_K(512)>;
+
+			fmu: flash-module@20000 {
+				ranges = <0x0 0x10000000 DT_SIZE_M(1)>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				compatible = "nxp,fmu-k4";
```

Comment:
Is this compatible chosen intentionally because flash isn't enabled in this PR? I don't see a YAML definition for it.

Reply:
yeah, I didn't put the flash controller enablement in this PR so far, but maybe I will add it since this PR is taking a while to get reviewed and it's pretty important for BLE enablement